stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -152,15 +152,23 @@ function (*)(A::AbstractMatrix, B::AbstractMatrix)
     TS = promote_op(matprod, eltype(A), eltype(B))
     mul!(similar(B, TS, (size(A,1), size(B,2))), A, B)
 end
-@inline mul!(C::StridedMatrix{T}, A::StridedVecOrMat{T}, B::StridedVecOrMat{T},
-             alpha::Union{T, Bool}, beta::Union{T, Bool}) where {T<:BlasFloat} =
-    gemm_wrapper!(C, 'N', 'N', A, B, MulAddMul(alpha, beta))
+
+@inline function mul!(C::StridedMatrix{T}, A::StridedVecOrMat{T}, B::StridedVecOrMat{T},
+                      alpha′::Number, beta′::Number) where {T<:BlasFloat}
+    alpha, beta = promote(alpha′, beta′, zero(T))
+    if alpha isa T && beta isa T
```

Comment:
Why not use a separate method for this?

Reply:
Are you suggesting to write

```julia
@inline function mul!(C::StridedMatrix{T}, A::StridedVecOrMat{T}, B::StridedVecOrMat{T},
                      alpha::$TA, beta::$TB) where {T<:BlasFloat}
    ...
```

with some expressions `$TA` and `$TB`?  Is it possible?  We need to express "alpha and beta would be promoted to `T`" in the type domain.  I don't know how to do it.